apps/desktop-wallet/src/pages/NewWallet/CheckWordsPage.tsx

```diff
@@ -1,325 +1,114 @@
-        newIndex -= 1
+  const handleOptionClick = (selectedWord: string) => {
+    const correctWord = mnemonicWords[currentWordIndex]
+    if (selectedWord === correctWord) {
+      if (currentWordIndex + 1 < mnemonicWords.length) {
+        setCurrentWordIndex(currentWordIndex + 1)
+        setOptions(generateOptions(mnemonicWords, currentWordIndex + 1))
+        setIsError(false)
+      } else {
+        sendAnalytics({ event: 'Creating wallet: Verifying words: Completed' })
+        onButtonNext()
       }
-
-      const filteredWords = selectedWords.filter((w) => w.key !== word.key)
-      setSelectedWords([...filteredWords.slice(0, newIndex), word, ...filteredWords.slice(newIndex)])
-      setClosestWordKey('')
+    } else {
+      setIsError(true)
     }
   }
 
-  // === Renders
-
-  const renderRemainingWords = () =>
-    wordList
-      .filter((w) => !selectedWords?.includes(w))
-      .map((w) => (
-        <RemainingWord onClick={() => setSelectedWords([...selectedWords, w])} key={w.key} layoutId={w.key}>
-          {w.word}
-        </RemainingWord>
-      ))
-
-  const renderSelectedWords = () =>
-    selectedWords?.map((w) => (
-      <SelectedWord
-        onPointerUp={() => handleSelectedWordRemove(w)}
-        key={w.key}
-        layoutId={w.key}
-        drag
-        ref={(element) => {
-          if (selectedElements && element) selectedElements[w.key] = element
-        }}
-        onDragStart={() => setIsDragging(true)}
-        onDrag={(e, info) => handleSelectedWordDrag(e, info, w, selectedElements)}
-        onDragEnd={() => handleSelectedWordDragEnd(w, closestWordKey)}
-      >
-        {isDragging && closestWordKey === w.key && <DragCursor layoutId="cursor" />}
-        {w.word}
-      </SelectedWord>
-    ))
-
-  const handleNextButtonPress = () => {
-    sendAnalytics({ event: 'Creating wallet: Verifying words: Clicked next' })
-    cleanup()
-    onButtonNext()
-  }
-
   const handleBackButtonPress = () => {
     sendAnalytics({ event: 'Creating wallet: Verifying words: Clicked back' })
-    cleanup()
     onButtonBack()
   }
 
-  const cleanup = () => {
-    setWordList([])
-    setSelectedElements({})
-    setSelectedWords([])
-  }
-
   return (
-    <FloatingPanel enforceMinHeight>
-      <PanelTitle color="primary" onBackButtonClick={onButtonBack} size="small">
-        {t('Security Check')}
-      </PanelTitle>
+    <FloatingPanel>
       <PanelContentContainer>
-        <Section>
-          <Paragraph>{t('Select the words in the right order.')}</Paragraph>
-          <SelectedWordList className={selectedWords.length === wordList.length ? (isValid ? 'valid' : 'error') : ''}>
-            {renderSelectedWords()}
-          </SelectedWordList>
-          <RemainingWordList>{renderRemainingWords()}</RemainingWordList>
+        <Section align="center">
+          <Paragraph>
+            <Trans
+              t={t}
+              i18nKey="Select the correct word for word number <1>{{ number }} of 24</1>."
```

Comment:
With the changes in this file, we block the user from importing a seed that has less than 24 words. The current production wallet allows the user to import a seed > 12 words which, to my knowledge, is possible through other Alephium wallets (node)